usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/WeightManagerClient.java

```diff
@@ -0,0 +1,201 @@
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(DELETE);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName + "/" + date,
+                accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Deletes all the weights of the specified pet from database.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @return The response code
+     * @throws ExecutionException When the deletion fails
+     * @throws InterruptedException When the deletion is interrupted
+     */
+    public int deleteAllWeight(String accessToken, String owner, String petName)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(DELETE);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName,
+                accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Gets a weight identified by its pet and date.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date the weight was created
+     * @return The WeightData identified by the data
+     */
+    public WeightData getWeightData(String accessToken, String owner, String petName, DateTime date)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder json = taskManager.execute(BASE_URL + owner + "/" + petName + "/" + date,
+                accessToken).get();
+        return GSON.fromJson(json.toString(), WeightData.class);
+    }
+
+    /**
+     * Gets the data from all the specified weights from the database identified by its pet.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @return The List containing all the meals from the pet
+     */
+    public List<Weight> getAllWeightData(String accessToken, String owner, String petName)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response = taskManager.execute(BASE_URL + owner + "/" + petName,
+                accessToken).get();
+        List<Weight> weightList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] weights = jsonArray.split(",\\{");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String ",\\{" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5361615)